design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,243 @@
+# Add Support for extra hardware parameters in Disk, RAM, CPU and NIC
+ 
+## Table of Contents
+ 
+<!--ts-->
+   * [Hardware Classification](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Investigation Details](#investigation-details)
+         * [Implementation Details/Notes/Constraints ](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigation](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+ 
+Implement support for more parameters in HWCC which will help to 
+classify a baremetal host on more possible configurations. 
+ 
+## Motivation
+
+We have hardware classification controller which validates and 
+classifies the hosts based on user hardware requirements.
+We would like to add more parameters which helps user to 
+identify the ideal host. This feature helps user to avoid runtime failures
+and also increases the performance for workload deployments.
+
+### Goals
+
+Automatically classify bare metal hosts based on user-provided rules
+in a way that other controllers can use, without having to evaluate
+all of those rules themselves.
+
+We will carry testing of HWCC only on Dell hardware. Testing of HWCC on 
+hardware from other vendors will not be goal of this proposal.
+
+## Proposal
+ 
+We are submitting proposal based on the issue raised in HWCC: 
+[Issue 53](https://github.com/metal3-io/hardware-classification-controller/issues/53)
+
+### Investigation Details
+
+Add support for new parameters:
+
+1. CPU
+
+    `arch` will come under CPU parameter.
+
+    User can provide `arch` type based on requirement.
+
+    Investigation Details: Below is sample introspection data for CPU.
+     `arch` parameter will be added under `cpu` section.
+    Types of architecture are 32-bit (x86) and 64-bit 
+    (x86-64, IA64, and AMD64).
+
+    ```yaml
+    hardware:
+        cpu:
+           arch: "x86_64"
+    ```
+
+2. Disk
+
+    `type` will come under disk parameter.
+
+    User can provide multiple disk type (RAID/HDD/SSD/NVMe), count and
+     individual size based on requirement.
```

Comment:
Some of the rules described here seem like they may be vendor or even host-specific. It might be better to expose the rotational and hctl fields, and let users define classification rules that interpret the values they find, than for us to hard-code some of those rules into a "type" value ourselves.

Reply:
You means to say, we have to define rotational and hctl field as a type instead of RAID, HDD, SSD, NVMe. or you want to keep both ?

Reply:
I mean to have the fundamental values we can discover as fields, and let the user combine them to create their own rules. So we would only have `rotational` and `hctl` fields (although maybe it would be better to use the same data structure that we use for `rootDeviceHints` on the BareMetalHost).